### PR TITLE
[#326] Auth system test

### LINKED ISSUE #326: Use authentication in system tests

### PR DESCRIPTION
**Pull request by FGRCL**
# General info
Add login to system tests.

**Issue number**: 

**Task description**: #326 

 - Add a login system test
 - Reuse the `AuthorizedTestRestTemplate` from integration tests
 
 
| Original   | Updated|
|:---------------:|:---------------:|
| | |

# Testing
1 new system tests for login
**Test coverage**:

# Additional notes
The system tests are now run as spring integration tests to leverage spring injection in the system tests. This means you shouldn't run the BE server before running the system tests, It'll start it on its own.

**Note to reviewers**:

**To do before merging**:
- [ ] Edit Changelog(if necessary)

### CODE REVIEW

**Review by ButlerJAL — CHANGES_REQUESTED** [review]
System test runs flawlessly. Change the code as suggested to pass Google Java Format check for approval.

File: src/main/java/org/beanpod/switchboard/config/WebMvcConfiguration.java [hunk 1]
```diff
@@ -13,7 +13,8 @@ public WebMvcConfigurer webMvcConfigurer() {
     return new WebMvcConfigurer() {
       @Override
       public void addCorsMappings(CorsRegistry registry) {
-        registry.addMapping("/**").allowedMethods("GET", "POST", "PUT", "DELETE");
+        registry.addMapping("/**")
+            .allowedMethods("*");
```
Comment (ButlerJAL) [inline src/main/java/org/beanpod/switchboard/config/WebMvcConfiguration.java]: Suggested change to pass Google Java Format.
```suggestion
        registry.addMapping("/**").allowedMethods("*");
```

Comment (FGRCL) [reply src/main/java/org/beanpod/switchboard/config/WebMvcConfiguration.java]: super!

**Review by ButlerJAL — APPROVED** [review]

**Review by felixlapierre — APPROVED** [review]
Works great! I love watching the system tests run 😍 it's so cool.

**Pull request merged**